### PR TITLE
Added compatibility for Blend Mode in `Light2D`/`PointLight2D` when converting from Godot 3 to 4.

### LINKED ISSUE #77887: --convert-3to4 doesn't upgrade Light2D Blend Mode `mode` -> `blend_mode`
### Godot version

v4.0.2.stable.official [7a0977ce2]

### System information

Linux Ubuntu 22.04 with Unity desktop

### Issue description

Upgrading a Godot 3 project to Godot 4 (via GUI or `godot4 --convert-3to4`) does not upgrade the Light2D `mode` member to `blend_mode`.

As a result, after opening the upgraded project, all non-Add blend modes have been lost. Subtract, Mix and Mask are all reset to Add blending mode.

Note: Mask has been removed and now requires CanvasItem clipping, which is a much more complex upgrade. I suppose we can print some warnings in the console (and if possible, in some popup when upgrading via GUI) to warn user that they need more work to upgrade this feature usage to Godot 4. It should be added on conversion doc side too, but this task has already been added on https://github.com/godotengine/godot-docs/issues/5121

At least Subtract and Mix should be upgraded correctly.

### Steps to reproduce

1. Add some lights with different blending modes in the project... or just download the MRP below
2. Run `godot4 --convert-3to4` or upgrade project via GUI
3. Observe that blending modes were lost.

Hint: for a quicker check without opening Godot, use the MRP which is under Git, run `godot4 --convert-3to4` in the console and check the Git diff to see that `mode` has not been renamed `blend_mode`.

### Minimal reproduction project

[v3.5.2 - convert 3to4 Light2D Mix Blending Mode reset to Add.zip](https://github.com/godotengine/godot/files/11655217/v3.5.2.-.convert.3to4.Light2D.Mix.Blending.Mode.reset.to.Add.zip)

### PR DESCRIPTION
**Pull request by TheSofox**
Fixes #77887

This ensures that when upgrading your project from Godot 3 to Godot 4, the Blend Mode of any `Light2D` (that becomes `PointLight2D`) will be preserved.

### CODE REVIEW

File: scene/2d/light_2d.cpp [hunk 1]
```diff
@@ -425,6 +425,15 @@ real_t PointLight2D::get_texture_scale() const {
 	return _scale;
 }
 
+bool PointLight2D::_set(const StringName &p_name, const Variant &p_value) {
```
Comment (akien-mga) [inline scene/2d/light_2d.cpp:428]: This should be enclosed in `#ifndef DISABLE_DEPRECATED`.

Comment (TheSofox) [reply scene/2d/light_2d.cpp:429]: Done.
Also enclosed the function declaration in the header file. Seemed the right thing to do but let me know if I need to change it.

Comment (akien-mga) [reply scene/2d/light_2d.cpp:429]: Yes I meant to do it for both :thumbup:

File: scene/2d/light_2d.cpp [hunk 2]
```diff
@@ -425,6 +425,17 @@ real_t PointLight2D::get_texture_scale() const {
 	return _scale;
 }
 
+#ifndef DISABLE_DEPRECATED
+bool PointLight2D::_set(const StringName &p_name, const Variant &p_value) {
+	if (p_name == "mode" && p_value.is_num()) { //Compatibility for Godot 3.
```
Comment (akien-mga) [inline scene/2d/light_2d.cpp:430]: I missed this earlier as I was reviewing on phone, there should be a space after `//`.
While changing it, might as well tweak language to match other similar 3.x compat checks.
```suggestion
	if (p_name == "mode" && p_value.is_num()) { // Compatibility with Godot 3.x.
```

Comment (TheSofox) [reply scene/2d/light_2d.cpp]: Done. Thanks.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**